tests/test-buttercup.el

```diff
@@ -613,27 +613,40 @@
 ;;; Spies
 
 (describe "The Spy "
-  (let (saved-test-function saved-test-command)
+  (let (saved-test-function
+        saved-test-command
+        saved-test-function-throws-on-negative)
     ;; We use `before-all' here because some tests need to access the
     ;; same function as previous tests in order to work, so overriding
     ;; the function before each test would invalidate those tests.
+    ;; Unfortunately there's no way to do this lexically, so these
+    ;; function definitions are leaked after the tests run.
     (before-all
       (setq saved-test-function (and (fboundp 'test-function)
                                      (symbol-function 'test-function))
             saved-test-command (and (fboundp 'test-command)
-                                    (symbol-function 'test-command)))
+                                    (symbol-function 'test-command))
+            saved-test-function-throws-on-negative
+            (and (fboundp 'test-function-throws-on-negative)
+                 (symbol-function 'test-function-throws-on-negative)))
       (fset 'test-function (lambda (a b)
                              (+ a b)))
       (fset 'test-command (lambda ()
                             (interactive)
-                            t)))
+                            t))
+      (fset 'test-function-throws-on-negative
+            (lambda (x) (if (>= x 0) x (error "x is less than zero")))))
     (after-all
       (if saved-test-function
           (fset 'test-function saved-test-function)
         (fmakunbound 'test-function))
       (if saved-test-command
           (fset 'test-command saved-test-command)
-        (fmakunbound 'test-command)))
+        (fmakunbound 'test-command))
+      (if saved-test-function-throws-on-negative
+          (fset 'test-test-function-throws-on-negative
+                test-function-throws-on-negative)
```

Comment:
And shouldn't this be `saved-test-function-throws-on-negative`?

Reply:
Fixed in 0460310.